src/event-engine/states/handshake_stopped.ts

```diff
@@ -1,20 +1,42 @@
+import { Cursor } from '../../models/Cursor';
 import { State } from '../core/state';
 import { Effects } from '../effects';
-import { Events, reconnect, subscriptionChange } from '../events';
+import { Events, reconnect, restore, subscriptionChange, unsubscribeAll } from '../events';
 import { HandshakingState } from './handshaking';
+import { UnsubscribedState } from './unsubscribed';
 
 type HandshakeStoppedStateContext = {
   channels: string[];
   groups: string[];
+  cursor?: Cursor;
 };
 
-export const HandshakeStoppedState = new State<HandshakeStoppedStateContext, Events, Effects>('STOPPED');
+export const HandshakeStoppedState = new State<HandshakeStoppedStateContext, Events, Effects>('HANDSHAKE_STOPPED');
 
-HandshakeStoppedState.on(subscriptionChange.type, (_context, event) =>
+HandshakeStoppedState.on(subscriptionChange.type, (context, event) =>
   HandshakeStoppedState.with({
     channels: event.payload.channels,
     groups: event.payload.groups,
+    cursor: context.cursor,
   }),
 );
 
-HandshakeStoppedState.on(reconnect.type, (context) => HandshakingState.with({ ...context }));
+HandshakeStoppedState.on(reconnect.type, (context, event) =>
+  HandshakingState.with({
+    ...context,
+    cursor: context.cursor || event.payload.cursor,
```

Comment:
Should we use this version, try to handle user-provided `cursor` before trying checking `cursor` from the context
```suggestion
    cursor: event.payload.cursor || context.cursor,
```

Reply:
updated wherever applicable ✔️